lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -181,20 +189,33 @@ data Listen
       -- ^ Listen on an unused TCP port, selected at random
     deriving (Show, Eq)
 
+-- | The type constraints of the API server.
+--
+--  * @s@ is the type of address discovery state.
+--  * @t@ is the type of the target (node backend).
+--  * @k@ is the key type, which determines the address derivation scheme.
+type WalletServer s t k =
```

Comment:
:+1: for the comment, which makes these type parameters a little easier to understand at a glance.